app/domain/authentication/constraints/permitted_constraint.rb

```diff
@@ -0,0 +1,23 @@
+module Authentication
+  module Constraints
+
+    # This constraint validates that only the initialized restrictions are permitted.
+    # So, no other restrictions can be found in the given restrictions when calling validate(given_restrictions).
+    class PermittedConstraint
+
+      def initialize(permitted:)
+        @permitted = permitted
+      end
+
+      def validate(resource_restrictions:)
+        (resource_restrictions - @permitted).each do |not_supported_restriction_name|
```

Comment:
consider splitting this and adding a line `not_supported_restrictions = resource_restrictions - @permitted` to ease readability

Reply:
done